lib/core/mqtt.cpp

```diff
@@ -99,9 +99,7 @@ mqtt::message_ptr Mqtt::messageToMessagePtr(const MqttMessage &message)
   header.AddMember("priority", message.header.priority, payload.GetAllocator());
   header.AddMember("timestamp", message.header.timestamp, payload.GetAllocator());
```

Comment:
also, idk if using `payload.GetAllocator()` with `header` is bad. I did it, but it might cause memory leaks/other weirdness so take a look if this is ok

Reply:
Not sure what best practice is here, cause there's no allocator for `rapidjson::Value`'s

Reply:
yeah, I think that was why I did it that way, but I'm pretty sure there should be a correct way to do it

Reply:
can you check this isn't leaking any memory jic